locust/test/subprocess_utils.py

```diff
@@ -0,0 +1,228 @@
+        )
+
+    # Check all output logs (stateless)
+    def _expect_any(self, to_expect, is_match: Callable[[Any, str], bool], *, stream="stderr"):
+        __tracebackhide__ = True
+
+        if stream == "stdout":
+            buffer = self.stdout_output
+        else:
+            buffer = self.stderr_output
+
+        if any(is_match(to_expect, line) for line in buffer):
+            return
+
+        self.on_fail(f"Did not see expected message: '{to_expect}'. Got {buffer[-5:]}")
+
+    def _not_expect_any(self, to_not_expect, is_match: Callable[[Any, str], bool], *, stream="stderr"):
+        __tracebackhide__ = True
+
+        if stream == "stdout":
+            buffer = self.stdout_output
+        else:
+            buffer = self.stderr_output
+
+        if any(is_match(to_not_expect, line) for line in buffer):
+            self.on_fail(f"Found unexpected message: '{to_not_expect}'.")
+
+    def expect(self, output: str, **kwargs):
+        __tracebackhide__ = True
+
+        is_match: Callable[[str, str], bool] = lambda out, line: out in line
+        return self._expect(output, is_match, **kwargs)
+
+    def expect_any(self, output: str, **kwargs):
+        __tracebackhide__ = True
+
+        is_match: Callable[[str, str], bool] = lambda out, line: out in line
+        return self._expect_any(output, is_match, **kwargs)
+
+    def not_expect_any(self, output: str, **kwargs):
+        __tracebackhide__ = True
+
+        is_match: Callable[[str, str], bool] = lambda out, line: out in line
+        return self._not_expect_any(output, is_match, **kwargs)
+
+    def expect_regex(self, pattern: str | re.Pattern[str], **kwargs):
+        __tracebackhide__ = True
+
+        if isinstance(pattern, str):
+            regex = re.compile(pattern)
+        else:
+            regex = pattern
+
+        is_match: Callable[[re.Pattern, str], bool] = lambda pattern, line: pattern.search(line) is not None
+        return self._expect(regex, is_match, **kwargs)
+
+    def expect_regex_any(self, pattern: str | re.Pattern[str], **kwargs):
+        __tracebackhide__ = True
+
+        if isinstance(pattern, str):
+            regex = re.compile(pattern)
+        else:
+            regex = pattern
+
+        is_match: Callable[[re.Pattern, str], bool] = lambda pattern, line: pattern.search(line) is not None
+        return self._expect(regex, is_match, **kwargs)
+
+    def send_input(self, content: str):
+        if self.use_pty:
+            os.write(self.stdin_m, content.encode())
+
+    def terminate(self):
+        if not IS_WINDOWS:
+            sig = signal.SIGINT
+        else:
+            # Signals are hard on Windows
+            sig = signal.SIGTERM
+
+        self.proc.send_signal(sig)
+        self._terminated = True
+
+    def wait(self, timeout=None):
+        return self.proc.wait(timeout=timeout or self.join_timeout)
+
+
+class PsutilTestProcess(TestProcess):
+    process_backend = psutil
+    proc: psutil.Popen  # type: ignore
```

Comment:
Hmm... this subclass feels weird, and process_backend isnt the most obvious thing. 

psutil.Process objects can be created from an existing subprocess though. So in the few/one case we need to list children the test case can just do:

        psutil.Process(proc.proc.pid).children(...)